server/src/main/java/org/apache/druid/server/metrics/ServiceStatusMonitor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.metrics;
+
+import com.google.common.base.Supplier;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.druid.java.util.metrics.AbstractMonitor;
+
+import java.util.Map;
+
+/**
+ * Monitor service running status.
+ * For Overlord/Coordinator, the metric reported is service leader count.
+ */
+public class ServiceStatusMonitor extends AbstractMonitor {
+
+  @Named("heartbeat")
+  @Inject(optional = true)
+  Supplier<Map<String, Number>> heartbeatTagsSupplier = null;
+
+  @Inject
+  public ServiceStatusMonitor() {
+  }
+
+  @Override
+  public boolean doMonitor(ServiceEmitter emitter) {
+    if (heartbeatTagsSupplier == null) {
+      return true;
+    }
+
+    heartbeatTagsSupplier.get().forEach((k, v) -> {
```

Comment:
If I am not mistaken, the tags should each be a separate dimension and not be emitted as metric values.
The metric value will always be 1, as it is a simple count.

In the current code, you would be emitting the `druid/heartbeat` metric multiple times in every invocation of `doMonitor`.

Reply:
Not sure I understand.
For Overlord/Coordinator, this `druid/heartbeat` metric will be 1 with `heartbeatType` dimension set to `leader`. And since this map `Map<String, Number>` only have one entry, so the metric only reported once per `doMonitor` call.
For other potential service (or component within the service) to use this monitor, `druid/heartbeat` metric doesn't have to be 1, that's why `heartbeatType` dimension is introduced as the metric can have different meaning for different `heartbeatType`.

Reply:
I think what Kashif is mentioning here is that `heartbeatTagsSupplier` should be a map of dimension keys to values and the metric that is reported for the heartbeat is always a constant.

Reply:
Thanks, updated the code with dimensions configured.